src/adagrams.js

```diff
@@ -1,7 +1,104 @@
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    // eslint-disable-next-line max-len
+    const letters = ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'B', 'B', 'C', 'C', 'D', 'D', 'D', 'D', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'F', 'F', 'G', 'G', 'G', 'H', 'H', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'J', 'K', 'L', 'L', 'L', 'L', 'M', 'M', 'N', 'N', 'N', 'N', 'N', 'N', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'P', 'P', 'Q', 'R', 'R', 'R', 'R', 'R', 'R', 'S', 'S', 'S', 'S', 'T', 'T', 'T', 'T', 'T', 'T', 'U', 'U', 'U', 'U', 'V', 'V', 'W', 'W', 'X', 'Y', 'Y', 'Z'];
+
+    const shuffleLetters = function(array) {
+      //used online resources to piece this together
+      let i = array.length - 1;
+      while (i > 0) {
+        const j = Math.floor(Math.random() * i);
+        const temp = array[i];
+        array[i] = array[j];
+        array[j] = temp;
+        i--;
+      }
+      return array;
+    };
+
+    const randomOrder = shuffleLetters(letters)
+
+    const createHand = function(array) {
+      const drawn = []
+      while (drawn.length < 10) {
+        const letter = array[0]
+        drawn.push(letter);
+        array.shift();
+      } 
+      return drawn;
+    };
+    return(createHand(randomOrder));
+     
   },
+
+
+  usesAvailableLetters(input, lettersInHand) {
+    const hand = {}
+    let outcome = true
+    lettersInHand.forEach(element => {
+      if (hand[element]) {
+        hand[element] += 1;
+      } else {
+        hand[element] = 1;
+      }
+    });
+  
+    const letters = input.split('')
+
+    letters.forEach(char => {
+      if (hand[char] < 0) {
+        outcome = false;
+      } else if (hand[char]) {
+        hand[char] -= 1;
+      } else {
+        outcome = false;
+      } 
+    });
+    return outcome;
+    
+  },
+
+  scoreWord(input) {
+    const letterPoints = {
```

Comment:
style note: move this object to ahead of all the functions, but inside the module.